test/unit/sdam/server_selection.test.js

```diff
@@ -31,6 +32,67 @@ describe('server selection', function () {
     isWritablePrimary: true,
     ok: 1
   });
+  const unknown = new ServerDescription('127.0.0.1:27022', {
+    ok: 0
+  });
+
+  describe('#sameServerSelector', function () {
+    const serverDescriptions = new Map();
+    serverDescriptions.set('127.0.0.1:27017', primary);
+    serverDescriptions.set('127.0.0.1:27018', unknown);
+
+    context('when the server is unknown', function () {
+      const topologyDescription = new TopologyDescription(
```

Comment:
could we move the `topologyDescription` and the list of servers (`serverList = Array.from(serverDescriptions.values())`) into a `beforeEach`? Since the same descriptions are used for all the test cases and only the input to sameServerSelector varies, it would make sense to isolate the rest of the shared setup, too

Reply:
Moved into `beforeEach` and `before` blocks for setup now.